sourcegraph/deploy-sourcegraph-k8s
# Add imagepullsecrets component with example

## Description

RE: https://sourcegraph.slack.com/archives/C02E4HE42BX/p1678721977364219

> We have a component in Kustomize for using a private registry, but it looks like that component doesn’t account for adding image pull secrets. Is the customer (and am I) missing something here?

User can add a patch to add new field to all resources, for example:

```yaml
patches:
  - patch: |-
      - op: add
        path: /spec/template/spec/imagePullSecrets
        value:
          name: registrykey
    target:
      group: apps
      kind: StatefulSet|Deployment|DaemonSet
      version: v1
```

This PR turns this suggested patch into a component for easier configuration.



---

## Checklist



- [ ] Update [CHANGELOG.md](https://github.com/sourcegraph/sourcegraph/blob/main/CHANGELOG.md)
- [ ] Update [K8s Upgrade notes](https://github.com/sourcegraph/sourcegraph/blob/main/doc/admin/updates/kubernetes.md)
- [x] Kustomiz-specific changes
- [ ] Update sister repository: [deploy-sourcegraph-helm](https://github.com/sourcegraph/deploy-sourcegraph-docker)
- [ ] Update sister repository: [deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker)
- [x] Verify all images have a valid tag and SHA256 sum

## Test plan



1. Run `kubectl kustomize examples/demo/private-registry -o test.yaml` from the root of this repository in this branch
2. check the output file `test.yaml`to see the `imagePullSecrets` field has been added to all resources with value from buildConfig.yaml
